presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/utils/FlightConnectorTestBase.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "arrow/flight/api.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/StaticFlightServer.h"
+#include "velox/common/config/Config.h"
+#include "velox/connectors/Connector.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+
+namespace facebook::presto::connector::afc::test {
+
+static const std::string kFlightConnectorId = "test-flight";
+
+class FlightConnectorTestBase : public velox::exec::test::OperatorTestBase {
+ public:
+  void SetUp() override;
+
+  void TearDown() override;
+
+ protected:
+  explicit FlightConnectorTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : config_{config} {}
+
+  FlightConnectorTestBase()
+      : config_{std::make_shared<velox::config::ConfigBase>(
+            std::move(std::unordered_map<std::string, std::string>{}))} {}
+
+ private:
+  std::shared_ptr<velox::config::ConfigBase> config_;
+};
+
+/// Creates and registers an arrow flight connector and
+/// spawns a Flight server for testing
+class FlightWithServerTestBase : public FlightConnectorTestBase {
```

Comment:
The reason I left a `With` in there is to avoid the false impression that this is used for testing a Flight server. It does not test the server - it tests the Flight connector, but also happens to spin up a server to help with the tests.

If you still feel it's better to change the name, we can rename to `FlightServerTestBase` as suggested, or maybe something different?